cli/azd/pkg/tools/azcli/azcli.go

```diff
@@ -78,6 +78,23 @@ type AzCli interface {
 		appName string,
 		deployZipFile io.Reader,
 	) (*string, error)
+	DeploySpringAppArtifact(
+		ctx context.Context,
+		subscriptionId string,
+		resourceGroup string,
+		instanceName string,
+		appName string,
+		relativePath string,
+		deploymentName string,
+	) (*string, error)
+	UploadSpringArtifact(
+		ctx context.Context,
+		subscriptionId string,
+		resourceGroup string,
+		instanceName string,
+		appName string,
+		artifactPath string,
+	) (*string, error)
```

Comment:
We have been trying to break away from this very large `AzCli` interface.  Instead please create a new struct/interface service specific for your spring apps functions (take a look at `cli\azd\pkg\tools\azcli\container_app.go` for an example.

Reply:
A little bit confuse about it. I already created `cli/azd/pkg/tools/azcol/spring_app.go`, it is same like `cli\azd\pkg\tools\azcli\container_app.go`.

Reply:
Yes, but we do not want these methods in the `AzCli` interface.  You should create your own `SpringService` interface instead.